StringIndexOutOfBoundsException in findClassesInPackage - Surefire/Maven - JDK 11 fails, JDK 8 succeeds.

### TestNG Version
7.1.0

### Expected behavior
TestNG reads the package name correctly, succeeds in doing a substring

### Actual behavior
TestNG keeps the package name from another class when doing a substring on current class, fails with error.

```
[ERROR] Caused by: java.lang.StringIndexOutOfBoundsException: begin 35, end 11, length 17
[ERROR]         at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
[ERROR]         at java.base/java.lang.String.substring(String.java:1874)
[ERROR]         at org.testng.internal.PackageUtils.findClassesInPackage(PackageUtils.java:112)
[ERROR]         at org.testng.xml.XmlPackage.initializeXmlClasses(XmlPackage.java:68)
[ERROR]         at org.testng.xml.XmlPackage.getXmlClasses(XmlPackage.java:59)
```

Having instrumented a copy of TestNG jar with print statements, I can see that this fails when a Jar has its files enumerated.  Specifically jaxb-api-2.3.1.jar, which include a file 'module-info.class', which lacks a package. PackageUtils then tries to do a substring based on the wrong package, with a class name that has no package, and fails.

This does not occur during the JDK 8 build, as Java 8 doesn't cause the JAXB jars to be referenced in this part of the build, but Java 11 does.

My simple print statement:
org.testng.internal.PackageUtils#findClassesInPackage
```
for (URL url : dirs) {
  String protocol = url.getProtocol();
  if (!matchTestClasspath(url, packageDirName, recursive)) {
    continue;
  }
  System.err.println("url = " + url);
```

With Java 8:
```
url = file:/C:/projects/testng-failure-case/target/test-classes/
url = file:/C:/projects/testng-failure-case/target/classes/
```

With Java 11:
```
url = file:/C:/projects/testng-failure-case/target/test-classes/
url = file:/C:/projects/testng-failure-case/target/classes/
url = jar:file:/C:/Users/jsnell/.m2/repository/org/apache/logging/log4j/log4j-api/2.11.2/log4j-api-2.11.2.jar!/META-INF/versions/9/
url = jar:file:/C:/Users/jsnell/.m2/repository/org/apache/logging/log4j/log4j-core/2.11.2/log4j-core-2.11.2.jar!/META-INF/versions/9/
url = jar:file:/C:/Users/jsnell/.m2/repository/javax/xml/bind/jaxb-api/2.3.1/jaxb-api-2.3.1.jar!/META-INF/versions/9/
```


### Is the issue reproductible on runner?

Completely reproducible with Maven test case using attached source.
See output:
[testng.failure.output.txt](https://github.com/cbeust/testng/files/4610452/testng.failure.output.txt)

### Test case sample

[testng-failure-case.zip](https://github.com/cbeust/testng/files/4610354/testng-failure-case.zip)

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,4 +1,5 @@
 Current
+Fixed: GITHUB-2308: StringIndexOutOfBoundsException in findClassesInPackage - Surefire/Maven - JDK 11 fails (Krishnan Mahadevan)
 Fixed: GITHUB:2788: TestResult.isSuccess() is TRUE when test fails due to expectedExceptions (Krishnan Mahadevan)
 Fixed: GITHUB-2800: Running Test Classes with Inherited @Factory and @DataProvider Annotated Non-Static Methods Fail (Krishnan Mahadevan)
 New: Ability to provide custom error message for assertThrows\expectThrows methods (Anatolii Yuzhakov)

--- a/testng-core-api/src/main/java/org/testng/internal/PackageUtils.java
+++ b/testng-core-api/src/main/java/org/testng/internal/PackageUtils.java
@@ -1,8 +1,9 @@
 package org.testng.internal;
 
+import static java.nio.charset.StandardCharsets.UTF_8;
+
 import java.io.File;
 import java.io.IOException;
-import java.io.UnsupportedEncodingException;
 import java.lang.reflect.Method;
 import java.net.JarURLConnection;
 import java.net.URL;
@@ -25,7 +26,6 @@ import org.testng.collections.Lists;
  * @author <a href="mailto:cedric@beust.com">Cedric Beust</a>
  */
 public class PackageUtils {
-  private static final String UTF_8 = "UTF-8";
   private static final String PACKAGE_UTILS = PackageUtils.class.getSimpleName();
   private static String[] testClassPaths;
 
@@ -34,11 +34,6 @@ public class PackageUtils {
 
   private PackageUtils() {
     // Utility class. Defeat instantiation.
-  }
-
-  /** Add a class loader to the searchable loaders. */
-  public static void addClassLoader(final ClassLoader loader) {
-    classLoaders.add(loader);
   }
 
   /**
@@ -96,6 +91,9 @@ public class PackageUtils {
         while (entries.hasMoreElements()) {
           JarEntry entry = entries.nextElement();
           String name = entry.getName();
+          if (name.startsWith("module-info") || name.startsWith("META-INF")) {
+            continue;
+          }
           if (name.charAt(0) == '/') {
             name = name.substring(1);
           }
@@ -183,11 +181,7 @@ public class PackageUtils {
     }
 
     String fileName = "";
-    try {
-      fileName = URLDecoder.decode(url.getFile(), UTF_8);
-    } catch (UnsupportedEncodingException ueex) {
-      // ignore. should never happen
-    }
+    fileName = URLDecoder.decode(url.getFile(), UTF_8);
 
     for (String classpathFrag : classpathFragments) {
       String path = classpathFrag + lastFragment;

